tests/test_storage.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+"""
+This is a unit test for representing storage in the MESSAGEix model, and
+testing the functionality of storage equations. The workflow is as follows:
+    - building a stylized MESSAGEix model
+    - adding seasonality and modifying parameters accordingly
+    - adding storage implementation
+    - testing storage functionality and equations
+
+"""
+
+from message_ix import Scenario
+
+# A flag to show if storage variables/parameters are added to ixmp java backend
+storage_in_java = False
+
+
+# A function for generating a simple MESSAGEix model
+def model_setup(scen, years):
+    scen.add_set('node', 'node')
+    scen.add_set('commodity', 'electr')
+    scen.add_set('level', 'level')
+    scen.add_set('year', years)
+    scen.add_set('type_year', years)
+    scen.add_set('technology', ['tec1', 'tec2'])
+    scen.add_set('mode', 'mode')
+    output_specs = ['node', 'electr', 'level', 'year', 'year']
+    dict_var_cost = {'tec1': 1, 'tec2': 2}
+
+    for yr in years:
+        scen.add_par('demand', ['node', 'electr', 'level', yr, 'year'],
+                     1, 'GWa')
+        for t in dict_var_cost.keys():
```

Comment:
everywhere you iterate through the keys, can be done like:

```
for tech, cost in dict_var_cost.items()
```

and then you can use the values explicitly

Reply:
and if you want to get **real** fancy, these loops can be done like

```
import itertools # up at the top
for year, (tech, cost) in itertools.product(years, dict_var_cost):
```

Reply:
Thanks a lot @Matt for the suggestions! I improved the code.